exasol_advanced_analytics_framework/udf_communication/peer_communicator/background_listener_interface.py

```diff
@@ -93,6 +93,11 @@ def register_peer(self, peer: Peer):
         register_message = messages.RegisterPeer(peer=peer)
         self._in_control_socket.send(serialize_message(register_message))
 
+    def send_payload(self, message: messages.Payload, payload: List[Frame]):
+        message = serialize_message(message)
+        frame = self._socket_factory.create_frame(message)
+        self._in_control_socket.send_multipart([frame] + payload)
```

Comment:
I'am a bit confused: I thought `out_control_socket` was for sending message within host between layers, and `in_control_socket` was for receiving messages within layers. Why do you now send to `in_control_socket`?

Reply:
Opposite of the same medal. This class lives on the Frontend and sends to the Backend. We talked until now only about the Backend. However, you are right the names are confusing. I think, we will need to inject also here a BackendSender and a BackendReciever to make this more clear and make this class also unit testable.